### DISCUSSION TITLE
Beta Launch Priority(DevOps)

### DISCUSSION THREAD

**Discussion by dwdraju** [icon-project/icon-bridge#106] [Ideas]
Devops priorities for Beta Launch readiness
1. Do all contracts compile or not? (Gradle refactoring ongoing by @sdpisreddevil )
    1. Javascore: Build, Test (Unit, Mock)
        1. bmc/ -> trigger build, optimizedJar, test
        2. bts/ -> trigger build, optimizedJar, test
    2. Solidity: Compile (all), Test (Unit, Mock)
        1. bmc/ -> trigger compile, test
        2. bts/ -> trigger compile, test
2. BMR code compile?
    1. Can build BMR docker image?
        1. cmd/iconbridge -> compilation test, unit/mock test 
    2. Can build e2etest(@manishbista28 )?
        1. cmd/e2etest -> compilation test, unit/mock test

**Repo Restructuring**
On the repo restructuring [branch](https://github.com/icon-project/icon-bridge/tree/repo_mgmt_and_github_actions), the following will be changes:
javascore -> contracts/javascore
solidity -> contracts/solidity
cmd/ -> bmr/cmd

**Comment by andrii-kl** [comment]
Do all contracts compile or not? (Gradle refactoring ongoing by @sdpisreddevil )

    Javascore: Build, Test (Unit, Mock)
        bmc/ -> trigger build, optimizedJar, test    -    Done
        bts/ -> trigger build, optimizedJar, test     -    Done
    Solidity: Compile (all), Test (Unit, Mock)
        bmc/ -> trigger compile, test          -    Done
        bts/ -> trigger compile, test             -    Done

https://github.com/icon-project/icon-bridge/issues/109

**Reply by dwdraju** [reply]
Thanks a lot @andrii-kl, I will split the steps that up on my branch's actions with rules.

**Comment by andrii-kl** [comment]
cmd/iconbridge -> compilation test  - Done
